### PR TITLE
Override host and cloud attributes in OTel Collector Agent

### PR DESCRIPTION
**Pull request by dmitryax**
Instrumentation libraries running in container environments cannot always correctly set host attributes while OTel Collector Agent has all required access to underlying host and can be considered as a source of truth for all host and cloud metadata. This change reconfigures Resource Detection processor on the agent to override all host and cloud attributes.

### CODE REVIEW

File: UPGRADING.md [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+
+If you still want to keep the previous behavior, use the following custom
+values.yaml configuration:
+
+```yaml
+agent:
+  config:
+    processors:
+      resourcedetection:
+        override: false
```
Comment (rmfitzpatrick) [inline UPGRADING.md:16]: Does the resource processor's new `upsert` prevent complete rollback?

Comment (rmfitzpatrick) [reply UPGRADING.md:16]: `--set 'agent.config.processors.resource.attributes[0].action=insert'` doesn't seem to properly merge:

```
      resource:
        attributes:
        - action: insert
```

Comment (dmitryax) [reply UPGRADING.md:16]: Not sure I understand what do you mean by complete rollback. `upsert` in resource processor is needed to keep existing behavior of setting `k8s.cluster.name` attribute -- it should be set from `clusterName` values.yaml config option, and `k8s.cluster.name` set by resourcedetector should be ignored.

Comment (dmitryax) [reply UPGRADING.md:16]: `k8s.cluster.name` is being set by resourcedetection processor only in some versions of GKE. Making sure this is taken care of in this rollback example, we would need to add something like:

```yaml
  config:
    processors:
      resourcedetection:
        override: false
      resource:
        attributes:
          - action: insert
            key: k8s.node.name
            value: "${K8S_NODE_NAME}"
          - action: upsert
            key: k8s.cluster.name
            value: < your cluster name from .Values.clusterName >
          < extra attribute from .Values.extraAttributes.custom >
```

I'm not sure likelihood of changed `k8s.cluster.name` warrants adding this config. We can probably just mention that `k8s.cluster.name` can be changed in GKE

Comment (dmitryax) [reply UPGRADING.md:16]: Hmm... Actually `upsert` is good. It doesn't need to be changed for the rollback. It preserves the required `k8s.cluster.name` value set from `clusterName` before and after the change.

**Review by rmfitzpatrick — APPROVED** [review]

**Pull request merged**